### PR TITLE
[py-lmfit] added version 1.0.2

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-lmfit/package.py [hunk 1]
```diff
@@ -12,17 +12,25 @@ class PyLmfit(PythonPackage):
     homepage = "http://lmfit.github.io/lmfit-py/"
     pypi = "lmfit/lmfit-0.9.5.tar.gz"
 
+    version('1.0.2',  sha256='67090ce56685cf7f92bd7358a1e7d4ad862b3758988109ec440e9825e5184b45')
     version('1.0.1',  sha256='d249eb756899360f4d2a544c9458f47fc8f765ac22c09e099530585fd64e286e')
     version('0.9.15', sha256='cd7bdf47c09a3d49f30dff9a1c7f778973d15d1e1b5dc642f14c22f6630eaf2f')
     version('0.9.5',  sha256='eebc3c34ed9f3e51bdd927559a5482548c423ad5a0690c6fdcc414bfb5be6667')
 
     depends_on('python@3.5:',               type=('build', 'run'), when='@1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lmfit/package.py]: Python 3.6+ is required now

File: var/spack/repos/builtin/packages/py-lmfit/package.py [hunk 2]
```diff
@@ -12,17 +12,25 @@ class PyLmfit(PythonPackage):
...
+    depends_on('py-asteval@0.9.16:',        type=('build', 'run'), when='@0.9.15:')
+    depends_on('py-asteval@0.9.21:',        type=('build', 'run'), when='@1.0.2:')
+
     depends_on('py-numpy@1.5:',             type=('build', 'run'), when='@0.9.5:')
     depends_on('py-numpy@1.16:',            type=('build', 'run'), when='@0.9.15:')
+    depends_on('py-numpy@1.18:',            type=('build', 'run'), when='@1.0.2:')
+
     depends_on('py-scipy@0.14:',            type=('build', 'run'), when='@0.9.5')
     depends_on('py-scipy@1.2:',             type=('build', 'run'), when='@0.9.15:')
+    depends_on('py-scipy@1.3:',             type=('build', 'run'), when='@1.0.2:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lmfit/package.py]: Are you sure these version reqs aren't from develop instead of 1.0.2?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-lmfit/package.py (hunk 2 above)
Comment (qwertos) [reply var/spack/repos/builtin/packages/py-lmfit/package.py]: These are pulled from the requirements.txt file in the tarball from pypi.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-lmfit/package.py]: `requirements.txt` means nothing to me, and it isn't used by pip or any other package manager that I know of.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-lmfit/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#important-files

Comment (qwertos) [reply var/spack/repos/builtin/packages/py-lmfit/package.py]: If that's the case, then the only dependency that I see is `setuptools`. `requirements.txt` is the only place where any other dependencies are listed.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-lmfit/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#python-dependencies

Comment (qwertos) [reply var/spack/repos/builtin/packages/py-lmfit/package.py]: Whoops. I don't know how I missed that the `setup.py` for version `1.0.2` actually had the dependencies listed. All earlier versions didn't. Fix incoming.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**